src/relion/cryolo_relion_it/icebreaker_histogram.py

```diff
@@ -0,0 +1,52 @@
+import gemmi
+import json
+import matplotlib.pyplot as plt
+from pathlib import Path
+import plotly.express as px
+import pandas as pd
+
+
+def create_json_histogram(working_directory):
+    icebreaker_path = str(working_directory / "External/Icebreaker_group/")
+    json_hist_file_name = "ice_hist.json"
+    if Path(icebreaker_path + "/particles.star").is_file():
+        data = extract_ice_column(icebreaker_path + "/particles.star")
```

Comment:
Suggest keeping path objects as path objects for as long as possible, avoid building paths using strings yourself, and only converting the objects right before a gemmi call. (in this case, add `os.fspath(...)` in the `extract_ice_column` function)
```suggestion
    icebreaker_path = working_directory / "External" / "Icebreaker_group"
    json_hist_file_name = "ice_hist.json"
    if icebreaker_path.joinpath("particles.star").is_file():
        data = extract_ice_column(icebreaker_path / "particles.star")
```